### ISSUE TITLE
[Bug]: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested

### ISSUE THREAD

**Issue by avivasyuta** [jekyll/jekyll#9536]
### Operating System

macOS 14.2.1 (23C71)

### Ruby Version

2.6.10p210 (2022-04-12 revision 67958) [universal.arm64e-darwin23]

### Jekyll Version

jekyll 4.2.2

### Expected Behavior

Jekyll image runs in a docker container on macBooks with Apple Silicon chips. I have m3 Pro chip.

### Current Behavior

Jekyll image run fails.

### Relevant log output

```shell
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
chown: _site/feed.xml: Permission denied
chown: _site/2024/01/10/single_number.html: Permission denied
chown: _site/2024/01/10: Permission denied
chown: _site/2024/01/10: Permission denied
...
```

### Code Sample

https://github.com/algorithmics-blog/algorithmics-blog.github.io

**Comment by alexohneander** [comment]
@avivasyuta can you tell me which image you are using?

**Comment by avivasyuta** [comment]
I use `jekyll/jekyll:4.2.2`.

Here is my make script

```
JEKYLL_VERSION=4.2.2

serve:
	docker run --rm \
	  --volume=$(PWD):/srv/jekyll \
	  -p 4000:4000 \
	  -it jekyll/jekyll:$(JEKYLL_VERSION) \
	  jekyll serve --livereload
```

**Comment by mkasberg** [comment]
The warning you're seeing is caused by running an amd64 Docker image on an M1 mac (or other ARM device) and is unrelated to your permission denied errors. You can add some options to your Docker command to make the warning go away if desired, but the warning isn't actually causing any problems.

The real problem is `chown: _site/feed.xml: Permission denied`. A filesystem permissions issue related to different user IDs (different owners) between the Docker container and your host machine. Possibly a new UID on your new M3 machine? Usage of `JEKYLL_UID` and `JEKYLL_GID` is documented in the [jekyll-docker README](https://github.com/envygeeks/jekyll-docker/blob/master/README.md#configuration).

[Docker entrypoint chown](https://github.com/envygeeks/jekyll-docker/blob/fb892998d444b7b2e4074adeb032197f67853c0a/repos/jekyll/copy/all/usr/jekyll/bin/entrypoint#L20-L36)

In short, this isn't a bug with Jekyll, and I think you can fix the problem with `JEKYLL_UID` (or perhaps just remove the `_site` dir on the host machine and allow it to be regenerated).
